### PR TITLE
APIS-3071 - Detects null value in path and returns a new ArrayEachNullError

### PR DESCRIPTION
**Pull request by fcantor**

### CODE REVIEW

File: parser.go [hunk 1]
```diff
@@ -914,6 +916,19 @@ func ArrayEach(data []byte, cb func(value []byte, dataType ValueType, offset int
...
+		// Detects nil in the path given
+		// for i, v := range data {
+		//fmt.Printf("index: [%d]   value: %s   address: [%v]\n", i, string(v), &v)
+		// if &i == nil {
+		// 	fmt.Println("YAY")
+		// 	return offset, ArrayEachNullError
+		// }
+		//}
+
+		if strings.Contains(string(data), "nil") {
```
Comment (kvincent2) [inline parser.go]: Would we want to loop over each item in the array so that we can provide a verbose error? I.e. key: [derp] is null

**Review by jstorer — CHANGES_REQUESTED** [review]
remove commented out code and add testing to demonstrate that what you added works as intended

File: parser_test.go [hunk 2]
```diff
@@ -1257,6 +1257,38 @@ func TestArrayEach(t *testing.T) {
...
+		cb   func(value []byte, dataType ValueType, offset int, err error)
+		keys []string
+	}
+	tests := []struct {
+		name       string
+		args       args
+		wantOffset int
+		wantErr    bool
+	}{
+		{"Array in second key's value with null", args{[]byte(`{"a":{"z":null,"b":[null],"c":null},"d":[]}`), funcError, []string{}}, 1, false},
```
Comment (jstorer) [inline parser_test.go]: formatting nit: break the fields into separate labeled lines to more resemble our other testing, for example
```
{
name: "stuff",
args: args,
etc....
}
```

File: parser_test.go [hunk 3]
```diff
@@ -1257,6 +1258,98 @@ func TestArrayEach(t *testing.T) {
...
+				data: []byte(`{"a":{"b":<HERE>["one","two"]},"c": null}`),
+				cb:   funcError,
+				keys: []string{"c"},
+			},
+			wantOffset: 36,
+			wantErr:    ArrayEachNullError,
+			wantCount:  0,
+		},
+		{
+			name: "Checking value in a, should error MalformedArrayError for missing subkey",
```
Comment (gannett-ggreer) [inline parser_test.go]: Nit: The subkey isn't missing, rather it is an object instead of an array.

Comment (fcantor) [reply parser_test.go]: Oh! Thanks, I'll make that correction.

**Review by gannett-ggreer — APPROVED** [review]

**Review by jstorer — APPROVED** [review]

**Pull request merged**